src/clj_fast/inline.clj

```diff
@@ -216,12 +216,40 @@
   Caution:
   For more than one path-value pair this macro will reorder code."
   [m & ksvs]
-  {:pre [(every? u/simple-seq? (take-nth 2 ksvs))]}
-  (lens/put-many
+  {:pre [(every? u/simple-seq? (take-nth 2 ksvs))
+         (apply distinct? (take-nth 2 ksvs))]}
+  (lens/update-many
    (fn [m k v] `(c/assoc ~m ~k ~v))
    (fn [m k] `(c/get ~m ~k))
    m ksvs))
 
+(defmacro update-in->
+  "Like update-in but inlines the calls when a static sequence of keys is
+  provided.
+  Can take an unlimited number of [ks exprs] pairs.
+  Unlike update-in, expressions are not & args, i.e.:
+  (update-many-in m [:a :b] f [:a :c] (g 1))
```

Comment:
Oh it looks like there's still a typo in the docstring here, update-many-in should be update-in->